src/frontend/src/optimizer/plan_node/logical_iceberg_scan.rs

```diff
@@ -141,9 +211,15 @@ impl PredicatePushdown for LogicalIcebergScan {
 
 impl ToBatch for LogicalIcebergScan {
     fn to_batch(&self) -> Result<crate::optimizer::plan_node::BatchPlanRef> {
-        let plan =
-            BatchIcebergScan::new(self.core.clone(), self.iceberg_scan_type, self.snapshot_id)
-                .into();
+        let iceberg_predicate =
+            rw_predicate_to_iceberg_predicate(self.predicate.clone(), self.schema().fields())?;
```

Comment:
If failed to convert it to iceberg predicate, we should keep it as a filter on top of iceberg scan instead of throwing error.